### ISSUE TITLE
Menu not loading in Edge

### ISSUE THREAD

**Issue by juliemars** [pymedusa/Medusa#4784]
**Describe the bug**
In todays update if I go to the webpage then the menu that should appear in the upper right hand corner is missing as can be seen in the below image.

![image](https://user-images.githubusercontent.com/17718528/43357957-f6db468e-9281-11e8-89c2-8e8e3e93ceca.png)

If I use Firefox then the menu appears.
![image](https://user-images.githubusercontent.com/17718528/43357952-dfea2dd2-9281-11e8-9322-f9a507b16500.png)

**To Reproduce**
1. Log into Medusa using Edge from latest Windows 10
2. Menu doesn't load. 

**Expected behavior**
Menu appears

**Edge Log**
I used F12 and under Console I have 8 errors shown:

![image](https://user-images.githubusercontent.com/17718528/43358004-b75e5202-9282-11e8-8582-88bfc9e0f4b5.png)

**Comment by OmgImAlexis** [comment]
Possible duplicate? https://github.com/pymedusa/Medusa/issues/4772

**Comment by sharkykh** [comment]
I remember trying it on the latest Edge and couldn't reproduce.
@juliemars Try <kbd>CTRL + F5</kbd>. Should eliminate the errors.

**Comment by juliemars** [comment]
Hi Sharkykh thanks for the reply, I tried that and nothing the issue remains. I have also cleared cookies and cache and tried on multiple PC's and the issue remains. Works fine in Firefox though but not private browsing.

**Comment by sharkykh** [comment]
You should try the `develop` branch then, maybe we inadvertently fixed the underlying issue.
Use git if you're familiar with it, or enter your Github credentials in the general config (under advanced tab), and you should be able to checkout the develop branch (option is above the Github credentials).

**Comment by sharkykh** [comment]
@OmgImAlexis 
I just noticed the Syntax Error in the `vue-native-websocket` library. You already moved that to Webpack, but I wonder why that is.

**Comment by juliemars** [comment]
@sharkykh and @OmgImAlexis  I had some spare time today and tried to troubleshoot this further. I think I have a cause but not a solution. 

I use a reverse proxy on NGINX to access the site if I navigate to the site using the internal IP, Port and base URL lets says http://127.0.0.1:8081/TV it works fine. 

I changed the base URL in Medusa and the reverse proxy and it still never worked which makes me think it is something in the reverse proxy setup I have causing the issue. 

I compared my proxy settings to the ones on https://github.com/pymedusa/Medusa/wiki/Reverse-Proxy-setup and the only difference was the top 2 lines mine has:

     location /TV/ {
        proxy_pass http://127.0.0.1:8081/TV/;

Which I expected, I am sure this is something simple but I cannot see what it is! if either of you have any ideas I would really appreciate it!

**Comment by juliemars** [comment]
Hi Sharkykh no worries hopefully someone else can help with the reverse proxy bit. 

When I navigate directly using the internal IP the syntax errors disappear 

I will update as soon as it is available and let you know if it fixes it.

**Comment by p0psicles** [comment]
Well I can confirm, but for me it's already not loading the login page.

```
HTML1300: Navigation occurred.
login
Current window: proxy.headshots.nl/medusa/login
SCRIPT5022: SyntaxError
build.js (110,36)
You are running Vue in development mode.
Make sure to turn on production mode when deploying for production.
See more tips at https://vuejs.org/guide/deployment.html
eval code (2048) (7751,7)
SCRIPT5007: Unable to get property 'fn' of undefined or null reference
vender.min.js (1,7696)
SCRIPT5009: 'jQuery' is undefined
bootstrap-formhelpers.min.js (6,1)
SCRIPT5009: '$' is undefined
parsers.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
index.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
init.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
notifications.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
init.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
popular-shows.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
recommended-shows.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
trending-shows.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
init.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
index.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
post-process.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
restart.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
status.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
failed-downloads.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
index.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
init.js (1,1)
SCRIPT5009: 'MEDUSA' is undefined
subtitle-missed.js (1,1)
SCRIPT5009: 'jQuery' is undefined
browser.js (1,2)
SCRIPT5009: 'PNotify' is undefined
notifications.js (3,1)
SCRIPT5007: Unable to get property 'push' of undefined or null reference
login (237,1)
SCRIPT5007: Unable to get property 'push' of undefined or null reference
login (503,1)
SCRIPT5009: 'Vue' is undefined
login (508,13)
SCRIPT5007: Unable to get property 'forEach' of undefined or null reference
app.js (55,1)
```

**Comment by sharkykh** [comment]
Check the first error:
```
SCRIPT5022: SyntaxError
build.js (110,36)
```
I think that's a webpack builtin. I wonder what is the syntax error.

**Comment by p0psicles** [comment]
I cant see the syntax error. Anything i can check?

**Comment by sharkykh** [comment]
@OmgImAlexis It's possible that's the case on `master`, but not on `develop`.
@p0psicles I'm not sure :\

**Comment by p0psicles** [comment]
Im running develop. The syntax highlighted piece of code is from that error.

**Comment by p0psicles** [comment]
I'm changing this to bug prio high. As medusa is just not working at all with it.
But we have edge listed as a supported browser.

So either we should fix this asap.
Or remove it from the supported browsers.

@pymedusa/developers thoughts?

**Comment by sharkykh** [comment]
IMO it should be fixed rather then dropping support for Edge.

@p0psicles 
Looks like it only affects reverse proxies and I don't have a setup to test this myself.
You can try commenting out the websocket code from [`./src/store/index.js`](https://github.com/pymedusa/Medusa/blob/develop/themes-default/slim/src/store/index.js#L72-L79) and rebuild the Webpack bundle.

**Comment by p0psicles** [comment]
I know that edge has troubles with unhandled promises or exceptions.
Maybe that's happening here, when using a reverse proxy?
I'm setting up a dev env on my win10 laptop.

**Comment by sharkykh** [comment]
@p0psicles It's possible, maybe the URL it tries to connect is incorrect, you should look for that when you're debugging.
And if you need it, this is the source of the minified code you sent earlier:
https://github.com/nathantsoi/vue-native-websocket/blob/v2.0.7/src/Observer.js#L21-L31

**Comment by p0psicles** [comment]
That didn't make any difference.
Drilled it down to this:

this is causing the error: `new WebSocket(e)`

Similar issue:
https://stackoverflow.com/questions/36667587/syntax-error-on-new-websocket-in-edge-internet-explorer

**Comment by p0psicles** [comment]
Yeah that fixed it. I'll clean up the branch and PR it.

**Comment by sharkykh** [comment]
@p0psicles 
I agree with you.
Good job tracking it down!

**Comment by juliemars** [comment]
Thanks @p0psicles for looking at it, I appreciate it.

Thanks everyone else for the help on this one.

**Comment by sharkykh** [comment]
Fixed in develop with #4928.
